### ISSUE TITLE
No errors when redefining elements of the same enum

### ISSUE THREAD

**Issue by Daniel-Cortez** [Zeex/pawn#597]
### Issue description:

Normally the compiler issues error 021 when the user tries to define two constants with the same name inside the same enum:
```Pawn
enum Enum1
{
	e0,
	e0 // error 021: symbol already defined: "e0"
};
```
But if there's another constant with the same name defined inside another named enum, for some reason the compiler doesn't print any errors, thus allowing redefinition:
```Pawn
enum Enum0 { e0 };
enum Enum1
{
	e0,
	e0 // no error(s)
};
```

### Minimal complete verifiable example (MCVE):

See above.

### Workspace Information:

* Compiler version: 3.10.4 - 3.10.10 (interestingly, earlier versions are not affected)
* Command line arguments provided (or sampctl version):
* Operating System:

**Comment by Daniel-Cortez** [comment]
OK, I think I've found the source of this bug.

As I mentioned in the above post, versions before 3.10.4 are not affected.
The main change in the 3.10.4 release was the addition of a new hashtable-based symbol search algorithm, which allowed to noticeably speed up the compilation process, but had a small downside to it: hashtable elements aren't ordered in the same way as in the global/local symbol tables. Before this change function `find_symbol()` (file `sc2.c`) traversed the global symbol table from the last defined symbol to the first one, so in situations when there were multiple enum elements with the same name, `find_symbol()` picked the one that was defined last, and after the adoption of the new search algorithm, `find_symbol()` couldn't guarantee this rule anymore.

This is why in the above example, in releases before 3.10.4 `find_symbol()` returns `Enum1::e0`, as it's the last defined symbol, but starting from v.3.10.4 `find_symbol()` returns `Enum0::e0` instead, as the new hashtable-based algorithm finds it first - this makes the compiler think the user tries to redefine a constant that belongs to another enum, so it allows the redefinition.

**Comment by Daniel-Cortez** [comment]
Fixed in #598, closing.
